src/contexts/ethereum-provider.context.tsx

```diff
@@ -26,24 +28,35 @@ const EthereumProviderProvider: FC = (props) => {
   const [provider, setProvider] = useState<Web3Provider>();
   const [account, setAccount] = useState<AsyncTask<string, string>>({ status: "pending" });
   const env = useEnvContext();
+  const { openSnackbar } = useGlobalContext();
 
   const connectProvider = async (walletName: WalletName): Promise<void> => {
     setAccount({ status: "loading" });
     switch (walletName) {
       case WalletName.METAMASK: {
         if (window.ethereum && window.ethereum.isMetaMask) {
           const web3Provider = new Web3Provider(window.ethereum);
-
           return getConnectedAccounts(web3Provider)
             .then((accounts) => {
               setProvider(web3Provider);
               setAccount({ status: "successful", data: accounts[0] });
             })
-            .catch(() => setAccount({ status: "failed", error: "Error loading account" }));
+            .catch((error) =>
+              parseError(error).then((errorMsg) => {
+                if (isMetamaskUserRejectedRequestError(error)) {
+                  setAccount({ status: "failed", error: errorMsg });
```

Comment:
In this case I wouldn't display any error, as it's not relevant for the user

Reply:
Right. Done.